CatAnalyzer/python/analyzers/ntuple_cff.py

```diff
@@ -0,0 +1,161 @@
+    )
+    process.ttllKin.jets = name+":jets"
+    process.ttllKin.leptons = "ttllLeptons"
+    process.ttllKin.met = name+":met"
+    process.ttllKin.metphi = name+":metphi"
+
+    #process.ntuple.floats.weights_genOthers  = cms.InputTag("flatGenWeights", "others")
+    process.ntuple.floats.kin_mLL = cms.InputTag("ttllKin", "mLL")
+    process.ntuple.floats.kin_mLB = cms.InputTag("ttllKin", "mLB")
+    #process.ntuple.floats.kin_mAddJJ = cms.InputTag("ttllKin", "mAddJJ")
+    #process.ntuple.floats.kin_dphi = cms.InputTag("ttllKin", "dphi")
+    process.ntuple.floats.kin_quality = cms.InputTag("ttllKin", "quality")
+
+    process.ntuple.cands.kin = cms.PSet(
+        src = cms.InputTag("ttllKin"),
+        exprsF = cms.untracked.PSet(
+            pt  = cms.string("pt"),
+            eta = cms.string("eta"),
+            phi = cms.string("phi"),
+            m   = cms.string("mass"),
+        ),
+    )
+    return process
+
+def ntupler_addVarsTTGen(process):
+    process.ntuple.ints.partonTop_modes = cms.InputTag("partonTop", "modes")
+
+    process.ntuple.cands.partonTop = cms.PSet(
+        src = cms.InputTag("partonTop"),
+        exprsF = cms.untracked.PSet(
+            pt  = cms.string("pt"),
+            eta = cms.string("eta"),
+            phi = cms.string("phi"),
+            m   = cms.string("mass"),
+        ),
+        exprsI = cms.untracked.PSet(
+            q = cms.string("charge"),
+            pdgId = cms.string("pdgId"),
+        ),
+    )
+    process.ntuple.cands.pseudoTop = cms.PSet(
+        src = cms.InputTag("pseudoTop"),
+        exprsF = cms.untracked.PSet(
+            pt  = cms.string("pt"),
+            eta = cms.string("eta"),
+            phi = cms.string("phi"),
+            m   = cms.string("mass"),
+        ),
+        exprsI = cms.untracked.PSet(
+            q = cms.string("charge"),
+            pdgId = cms.string("pdgId"),
+        ),
+    )
+
+    return process
+
+def ntupler_addVarsGen(process, name):
+    process.ntuple.float.weight = cms.InputTag(name, "weight")
+    process.ntuple.float.weight_gen = cms.InputTag("flatGenWeights", "")
+    process.ntuple.float.weight_pu   = cms.InputTag("pileupWeight")
+    process.ntuple.float.weight_puUp = cms.InputTag("pileupWeight", "up")
+    process.ntuple.float.weight_puDn = cms.InputTag("pileupWeight", "dn")
+
+    from CATTools.CatAnalyzer.csvWeights_cfi import csvWeights
+    setattr(process, 'csvWeights'+name, csvWeights.clone(src = cms.InputTag(name, "jets")))
+    del csvWeights
+    process.ntuple.float.weight_CSV = cms.InputTag("csvWeights"+name)
+    process.ntuple.floats.weights_CSV = cms.InputTag("csvWeights"+name, "syst")
+
+    process.ntuple.floats.weights_pdf = cms.InputTag("flatGenWeights", "pdf")
+    process.ntuple.floats.weights_scaleUp = cms.InputTag("flatGenWeights", "scaleup")
+    process.ntuple.floats.weights_scaleDn = cms.InputTag("flatGenWeights", "scaledown")
+
+    return process
+
+def ntupler_addVarsGenTop(process):
```

Comment:
This part is for the genTop. I'm not sure that these are the full set of variables for the analysis.